File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.js
class Solution {
  /**  @typedef {{start:number,end:number}} Interval */
  /** @param {Interval[]} intervals @returns {number} */
  minMeetingRooms(intervals) {
    const starts = intervals.map(({ start }) => start).sort((a, b) => a - b);
    const ends = intervals.map(({ end }) => end).sort((a, b) => a - b);
    let [count, res, s, e] = Array(4).fill(0);
    while (s < intervals.length) {
      if (starts[s] < ends[e]) {
        count++;
        s++;
      } else {
        count--;
        e++;
      }
      res = Math.max(res, count);
    }
    return res;
  }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l,r=0,0
        res=0
        if len(s)==1:
            return 1
        while r<len(s):
            while len(s[l:r+1])!=len(set(s[l:r+1])):
                l+=1
            r+=1
            res=max(res,r-l)
        return res